lnwire/closing_signed.go

```diff
@@ -55,9 +165,40 @@ var _ Message = (*ClosingSigned)(nil)
 //
 // This is part of the lnwire.Message interface.
 func (c *ClosingSigned) Decode(r io.Reader, pver uint32) error {
-	return ReadElements(
+	err := ReadElements(
 		r, &c.ChannelID, &c.FeeSatoshis, &c.Signature, &c.ExtraData,
 	)
+	if err != nil {
+		return err
+	}
+
+	// Attempt to parse out the FeeRange record.
+	var fr FeeRange
+	typeMap, err := c.ExtraData.ExtractRecords(&fr)
+	if err != nil {
+		return err
+	}
+
+	// Only set the FeeRange record if the TLV type was included in the
+	// stream.
+	if val, ok := typeMap[FeeRangeType]; ok && val == nil {
+		// Check that the fee range is sane before setting it.
+		if fr.MinFeeSats > fr.MaxFeeSats {
```

Comment:
Hmm, I think we might want to handle this check at a higher layer, particularly given that a concrete error isn't returned here, so the `readHandler` in the peer can't distinguish from violated constraints vs a garbled message (unparseable).

Reply:
IMO this check belongs here, it checks that the message is sane. We'd need a new function to check sanity otherwise, which accomplishes the same thing but requires another call in the chancloser

Reply:
why not have the check in the `case closeFeeNegotiation:` case in `ProcessCloseMsg` function where we also do the validation of other close messages? Tend to agree that this Decode message should only decode & not validate the meanings of the fields.